src/main/java/ro/msg/learning/shop/strategy/impl/SingleLocationStrategy.java

```diff
@@ -1,4 +1,116 @@
+            return stock.getQuantity() >= quantity;
+        }
+        return false;
+    }
+
+    private Product getProductFromPair(ProductQuantityPairUtil pair) {
+        UUID productId = pair.getProductId();
+        return productService.getProduct(productId);
+    }
+
+    private Stock getStock(Product product, Location location) {
+        StockId id = new StockId(product, location);
+        return stockService.getStock(id);
+    }
+
+    private Location verifyStock(Set<ProductQuantityPairUtil> productQuantityPairUtilList) {
+        Collection<Location> locationList = locationService.getLocations();
+        for (ProductQuantityPairUtil productQuantityPairUtil : productQuantityPairUtilList) {
+            Product product = getProductFromPair(productQuantityPairUtil);
+            locationList.removeIf(location -> !isProductOnLocation(location, product, productQuantityPairUtil.getQuantity()));
+        }
+        Optional<Location> optionalLocation = locationList.stream().findFirst();
+        return optionalLocation.orElse(null);
+    }
+
+    private void subtractStock(Location location, ProductQuantityPairUtil productQuantityPairUtil) {
+        Product product = getProductFromPair(productQuantityPairUtil);
+        Stock stock = getStock(product, location);
+        int quantity = stock.getQuantity() - productQuantityPairUtil.getQuantity();
+        Stock updatedStock = new Stock(product, location, quantity);
+        stockService.updateStock(updatedStock);
+    }
+
+    private void subtractAllStocks(Location location, Set<ProductQuantityPairUtil> productQuantityPairUtilList) {
+        for (ProductQuantityPairUtil productQuantityPairUtil : productQuantityPairUtilList) {
+            subtractStock(location, productQuantityPairUtil);
+        }
+    }
+
+    private Set<OrderDetail> createOrderDetails(Orders order, Location location, Set<ProductQuantityPairUtil> productQuantityPairUtilList) {
+        Set<OrderDetail> orderDetails = new HashSet<>();
+        for (ProductQuantityPairUtil productQuantityPairUtil : productQuantityPairUtilList) {
+            Product product = getProductFromPair(productQuantityPairUtil);
+            int quantity = productQuantityPairUtil.getQuantity();
+            OrderDetail orderDetail = new OrderDetail(order, product, location, quantity);
+            orderDetailService.createOrderDetail(orderDetail);
+            orderDetails.add(orderDetail);
+        }
+        return orderDetails;
+    }
+
+    @Override
+    public Set<OrderDetailsUtil> handleOrderCreation(Orders order, Set<ProductQuantityPairUtil> productQuantityPairUtilList) {
+        Customer customer = customerService.getCustomer(order.getCustomer().getId());
+        Location location = verifyStock(productQuantityPairUtilList);
+        Set<OrderDetailsUtil> orderDetailsUtilSet = new HashSet<>();
+
+        if (location != null) {
+            order.setCustomer(customer);
+            Orders newOrder = ordersRepository.save(order);
```

Comment:
Move all the logic for Orders in your OrdersService. The Strategy is strictly used to find the available locations for each product. Saving the order, subtracting the stock, and so on - this logic should be implemented in the service. The same applies to the other strategy.